server/src/main/java/io/crate/execution/dml/upsert/UpdateToInsert.java

```diff
@@ -244,26 +237,9 @@ public IndexItem convert(Doc doc, Symbol[] updateAssignments, Object[] excludedV
             }
             Maps.mergeInto(source, targetPath.name(), targetPath.path(), value);
         }
-
-        for (int pkIndex = 0; pkIndex < pkSize; pkIndex++) {
-            ColumnIdent pk = table.primaryKey().get(pkIndex);
-            Object value;
-            if (pk.isRoot()) {
-                int valuesIdx = Reference.indexOf(columns, pk);
-                assert valuesIdx > -1 : "Primary key column must exist in columns";
-                value = insertValues[valuesIdx];
-            } else {
-                int valuesIdx = Reference.indexOf(columns, pk.getRoot());
-                Object object = insertValues[valuesIdx];
-                value = Maps.getByPath((Map<String, Object>) object, pk.path());
-            }
-            if (value != null) {
-                primaryKeys[pkIndex] = value.toString();
-            }
-        }
         return new IndexItem.StaticItem(
             doc.getId(),
-            Arrays.asList(primaryKeys),
+            Id.decode(table.primaryKey(), doc.getId()),
```

Comment:
Initially I thought I could just return the docIds from the original indexItem - given that a primary key cannot be changed with an update. But we don't set the primaryKey values for updates:

https://github.com/crate/crate/blob/e23711ed2e3e0c8a6cb6326bc5c59b92f08bc844/server/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java#L318

So I opted to decode the id, which contains the values. Otherwise we'd need to include default expressions/generated column evaluation within `UpdateToInsert`